source/dofs/dof_tools_constraints.cc

```diff
@@ -1723,13 +1723,49 @@ namespace DoFTools
   }
 
 
+  namespace internal
+  {
+    template <int dim, int spacedim>
+    void check_if_fe_okay(const FiniteElement<dim,spacedim> &fe)
+    {
+      Assert(fe.get_name().find("FE_Q_Hierarchical<") == std::string::npos,
+             ExcNotImplemented("Constraints for FE_Q_Hierarchical are currently not implemented!"));
+    }
+
+
+    template <int dim, int spacedim>
+    void check_if_hanging_nodes_okay(const DoFHandler<dim,spacedim> &dof_handler)
+    {
+      check_if_fe_okay(dof_handler.get_fe());
+    }
+
+    template <int dim, int spacedim>
+    void check_if_hanging_nodes_okay(const hp::DoFHandler<dim,spacedim> &dof_handler)
+    {
+      for (unsigned int idx = 0; idx < dof_handler.get_fe().size(); ++idx)
+        {
+          check_if_fe_okay(dof_handler.get_fe()[idx]);
+        }
+    }
+  }
+
 
 
   template <typename DoFHandlerType>
   void
   make_hanging_node_constraints (const DoFHandlerType &dof_handler,
                                  ConstraintMatrix     &constraints)
   {
+#ifdef DEBUG
+    {
+      // blacklist FE_Q_Hierarchical, see:
+      // - https://github.com/dealii/dealii/issues/4767
+      // - tests/fail/project_q_hierarchical_03
+      // - tests/fe/mass_constraints
+      internal::check_if_hanging_nodes_okay (dof_handler);
```

Comment:
Could you add a short description of the problem? I don't like linking to GitHub issues as documentation since, like google code, one day we won't have access to this platform.